test/gtest/ucp/test_ucp_request.cc

```diff
@@ -100,3 +103,478 @@ UCS_TEST_P(test_ucp_request, test_request_query)
+                               (uint64_t)m_sbuf.data(), m_rkey, &param);
+            break;
+
+        case RMA_PUT:
+            sreq = ucp_put_nbx(sender().ep(), m_rbuf.data(), m_msg_size,
+                               (uint64_t)m_sbuf.data(), m_rkey, &param);
+            break;
+
+        case STREAM:
+            param.op_attr_mask = UCP_OP_ATTR_FIELD_DATATYPE;
+            param.datatype     = ucp_dt_make_contig(1);
+            sreq = ucp_stream_send_nbx(sender().ep(), m_sbuf.data(), m_msg_size,
+                                       &param);
+            break;
+
+        case AM:
+            sreq = send_am();
+            break;
+        }
+
+        ASSERT_TRUE(UCS_PTR_IS_PTR(sreq));
+        wait_and_restart(sreq, rreq, output_proto);
+        wait_receive(op);
+
+        request_wait(sreq);
+        request_wait(rreq);
+        flush_ep(sender());
+
+        EXPECT_EQ(m_sbuf, m_rbuf);
+    }
+
+    virtual void
+    wait_and_restart(void *sreq, void *rreq, const std::string &expected_proto)
+    {
+        ucp_request_t *req                 = (ucp_request_t*)sreq - 1;
+        const ucp_datatype_iter_t *dt_iter = &req->send.state.dt_iter;
+
+        while (dt_iter->offset == 0) {
+            progress();
+        }
+
+        EXPECT_LT(dt_iter->offset, dt_iter->length);
+        ASSERT_UCS_OK(ucp_ep_restart_nb(sender().ep()));
+
+        uint8_t *op_id_flags =
+                (uint8_t*)&req->send.proto_config->select_param.op_id_flags;
+        wait_for_value(op_id_flags, (uint8_t)(*op_id_flags |
+                                              UCP_PROTO_SELECT_OP_FLAG_RESUME));
+        EXPECT_STREQ(expected_proto.c_str(),
+                     req->send.proto_config->proto->name);
+    }
+
+    void init_rkey()
+    {
+        const ucp_mem_map_params_t params = {
+            .field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                          UCP_MEM_MAP_PARAM_FIELD_LENGTH,
+            .address    = m_sbuf.data(),
+            .length     = m_sbuf.size()
+        };
+
+        ASSERT_UCS_OK(ucp_mem_map(receiver().ucph(), &params, &m_memh));
+
+        void *rkey_buffer;
+        size_t rkey_buffer_size;
+        ASSERT_UCS_OK(ucp_rkey_pack(receiver().ucph(), m_memh, &rkey_buffer,
+                                    &rkey_buffer_size));
+
+        ASSERT_UCS_OK(ucp_ep_rkey_unpack(sender().ep(), rkey_buffer, &m_rkey));
+        ucp_rkey_buffer_release(rkey_buffer);
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants,
+                               UCP_FEATURE_TAG | UCP_FEATURE_RMA |
+                                       UCP_FEATURE_STREAM | UCP_FEATURE_AM,
+                               0, "");
+    }
+
+    static const size_t mb_to_bytes = 1048576;
+    size_t m_msg_size;
+    ucp_mem_h m_memh;
+    ucp_rkey_h m_rkey;
+    std::vector<char> m_sbuf;
+    std::vector<char> m_rbuf;
+};
+
+#ifdef ENABLE_DEBUG_DATA
```

Comment:
why some tests below don't work without debug data but `reset_protocol(TAG, "rndv/am/bcopy");` does work? can we enable all tests unconditionally?

Reply:
actually it's needed for 2 tests: rndv_get and atp (I moved it to cover only those tests).
Those tests need to access `req->recv.proto_rndv_request` field, which is only present under DEBUG_DATA

Reply:
ok, pls add the comment